runtime/lua/vim/lsp/util.lua

```diff
@@ -149,16 +149,42 @@ function M.apply_text_edits(text_edits, bufnr)
     table.insert(lines, '')
   end
 
+  -- store current cursor pos
+  local curpos_fixed = false
+  local curpos = api.nvim_win_get_cursor(0)
+  curpos[1] = curpos[1] - 1
```

Comment:
I personally find it a little difficult to track whether it's 1-index or 0-index when we use the same variable name. Can you use a different variable here to store the 0-indexed stuff and then add the 1 back at the end? I know it doesn't actually change anything, but it would make me feel better :laughing: 

Additionally, if you wanted to write a little local helper function here to keep track of these, that'd be be very clarifying as well.

Reply:
I've changed to use nvim's cursor pos to LSP's Position in here.